### PR TITLE
add create v4 keyfile

### LINKED ISSUE #26: Extend keyfile to handle eth2.0 BLS keying material
In particular, implement this:

https://eips.ethereum.org/EIPS/eip-2335

### PR DESCRIPTION
**Pull request by pacrob**
### What was wrong?

We can decode a v4 keyfile, but not create one

Closes #26 

### How was it fixed?

Add ability to create the v4 keyfile, plus tests and a note in the docs regarding maximum valid private keys.

Also added a `.readthedocs.yaml` in preparation for moving docs from the README to RTD

### Todo:

- [x] Clean up commit history
- [x] Add or update documentation related to these changes
- [x] Add entry to the [release notes](https://github.com/ethereum/eth-keyfile/blob/main/newsfragments/README.md)

#### Cute Animal Picture

image

### CODE REVIEW

File: eth_keyfile/keyfile.py [hunk 1]
```diff
@@ -127,13 +160,19 @@ def _create_v3_keyfile_json(
...
     password: bytes,
     kdf: KDFType,
-    work_factor: Union[int, None] = None,
-    salt_size: int = 16,
+    work_factor: Optional[int] = None,
+    salt_size: Optional[int] = None,
 ) -> Dict[str, Any]:
-    salt = Random.get_random_bytes(salt_size)
-
+    # fill in the blanks
```
Comment (kclowes) [inline eth_keyfile/keyfile.py:166]: remove comment?

File: setup.py [hunk 2]
```diff
@@ -48,8 +48,9 @@
     url="https://github.com/ethereum/eth-keyfile",
     include_package_data=True,
     install_requires=[
-        "eth-utils>=2",
         "eth-keys>=0.4.0",
+        "eth-utils>=2",
+        "py_ecc>=5.2.0",
```
Comment (kclowes) [inline setup.py:53]: This has to be `>=5.2.0`, right? We can't open it up more than that?

**Review by kclowes — APPROVED** [review]
I left a nit and a comment around the py_ecc dep, but looks good otherwise. Are there any other libraries that we can fuzz against here?

File: eth_keyfile/keyfile.py (hunk 1 above)
Comment (pacrob) [reply eth_keyfile/keyfile.py]: done!

File: setup.py (hunk 2 above)
Comment (pacrob) [reply setup.py:53]: Didn't check earlier, but `5.2.0` is from 2021 with max listed of py37, so figured it was old enough

**Comment by pacrob** [comment]
> Are there any other libraries that we can fuzz against here?

The `deposit-cli` tool is the implementation listed in the EIP. It doesn't look like that's on pypi, so for CI would involve pulling it from github or including a built binary. I can timebox adding that if you like.

The EIP lists TypeScript and Go versions too, but haven't looked at those.

**Comment by kclowes** [comment]
> CI would involve pulling it from github or including a built binary.

I don't think we need to put it on CI, but would be good to check that box manually if possible (within some timebox) 🤷

**Review by reedsa — APPROVED** [review]
Tests and notes look good 💯 
lgtm!

**Comment by pacrob** [comment]
> I don't think we need to put it on CI, but would be good to check that box manually if possible (within some timebox) 🤷

Okey! Have manually validated that an `eth-keyfile`-generated keystore dict can be loaded by a `staking-cli` `Keystore` dataclass (and that when the dataclass is `dict`ed, they match). In manually testing, I ran into the `salt` value sometimes being bytes instead of a hexstring, so I added a test to validate that the types of all our dict.values are the same as the EIP examples.

**Review by kclowes — APPROVED** [review]
LGTM! Nice work! 🚢
